client/src/components/Sub_ProductDetail/AddToCart.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import SizeSelector from './Sub_AddToCart/SizeSelector.jsx';
+import QuantitySelector from './Sub_AddToCart/QuantitySelector.jsx';
+
+function AddToCart({ skus, startSelect, currentStyle }) {
+  const [skuID, setSkuID] = useState();
+  const [quantity, setQuantity] = useState(1);
+  const [showAlert, setShowAlert] = useState(false);
+  const [validQuantities, setValidQuantities] = useState([1]);
+  const validSkus = Object.entries(skus).filter((entry) => entry[1].quantity > 0);
+
+  const handleSize = (e) => { setSkuID(e.target.value); setQuantity(1); };
+
+  const handleQuantity = (e) => setQuantity(e.target.value);
+
+  const submitCart = () => {
+    const config = {
+      sku_id: skuID,
+      count: quantity,
+    };
+    axios.post('/api/cart', config)
+      .then()
+      .catch((err) => console.log(err));
+  };
+
+  const handleClick = () => {
+    if (!skuID) {
+      setShowAlert(true);
+    } else {
+      submitCart();
+    }
+  };
+
+  useEffect(() => {
+    if (skuID) {
+      setValidQuantities(Array.from({ length: skus[skuID].quantity }, (_, i) => i + 1));
```

Comment:
say if `skus[skuID].quantity` was = `4`
`Array.from({ length: skus[skuID].quantity }, (_, i) => i + 1)` `// returns an array of [1, 2, 3, 4]`. 

am I understanding this right?
and are you using `_` in the arrow function parameter to indicate that the value is undefined?

Reply:
i use Array.from() function to generate an array with a range of numbers starting from 1 to the quantity of the stock items. the underscore _ is a placeholder for unused values. it's the value of the current element for the current iteration, and i is the index of the current iteration. the function (_, i) => i + 1) return 'i + 1', which means the value of each element in the final array will be the value of the index of the current iteration, incremented by 1, because i want the final array to start with 1 rather than 0.

Reply:
i use Array.from() function to generate an array with a range of numbers starting from 1 to the quantity of the stock items. the underscore _ is a placeholder for unused values. it's the value of the current element being mapped, and i is the index of the current iteration. the function (_, i) => i + 1) return 'i + 1', which means the value of each element in the final array will be the value of the index of the current iteration, incremented by 1, because i want the final array to start with 1 rather than 0.

Reply:
Very nice way to generate that array. I'm gonna keep that in mind.